src/Bridges/Constraint/bridges/norm_special_case.jl

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    NormSpecialCaseBridge{T,F} <: Bridges.Constraint.AbstractBridge
+
+`NormSpecialCaseBridge` implements the following reformulations:
+
+  * ``(t, x) in NormOneCone(d)`` into ``(t, x) in NormCone(1, d)``
+  * ``(t, x) in SecondOrderCone(d)`` into ``(t, x) in NormCone(2, d)``
+  * ``(t, x) in NormInfinityCone(d)`` into ``(t, x) in NormCone(Inf, d)``
```

Comment:
I've added this bridge for now, mainly to test out, because I'm not sure if this would be used by an actual solver. (I can't see a solver supporting `NormCone` or `PowerCone` but not `SecondOrderCone` or linear constraints.)

The other option I considered was the reverse. `NormCone(1, d)` creates `NormOneCone(d)`, etc.

That's a bit weird though, because the `NormCone` constructor wouldn't necessarily return a `::NormCone`. And some solver might support a general `NormCone` but not the specialized `NormOne` and `NormInfinity` cones.

It's a little messy.

Reply:
Yes, I wouldn't have the constructor do that. At least not at the MOI level.